### PR TITLE
Patch donation functional test

### PR DESCRIPTION
**Pull request by nickhsine**
@babygoat please help to review.

### CODE REVIEW

File: tests/controller_donations_test.go [hunk 1]
```diff
@@ -149,128 +149,130 @@ func testDonationDataValidation(t *testing.T, path string, authorization string)
...
+			Amount: testAmount,
+			Cardholder: cardholder{
+				Email: "developer@twreporter.org",
+			},
+		}
+
+		reqBodyInBytes, _ = json.Marshal(reqBody)
+		resp = serveHTTP("POST", path, string(reqBodyInBytes), "application/json", authorization)
+
+		assert.Equal(t, 400, resp.Code)
```
Comment (babygoat) [inline tests/controller_donations_test.go]: 400 -> http.StatusBadRequest

File: controllers/donation.go [hunk 2]
```diff
@@ -384,13 +385,17 @@ func (mc *MembershipController) PatchADonationOfAUser(c *gin.Context, donationTy
...
-	if err = mc.Storage.UpdateByConditions(map[string]interface{}{
+	if err, rowsEffected = mc.Storage.UpdateByConditions(map[string]interface{}{
 		"user_id": userID,
 		"id":      recordID,
 	}, d); err != nil {
 		return 0, gin.H{}, err
 	}
 
+	if rowsEffected == 0 {
+		return http.StatusNotFound, gin.H{"status": "error", "message": fmt.Sprintf("record(id: %d) not found", recordID)}, nil
```
Comment (babygoat) [inline controllers/donation.go]: In jsend format, client side error (4xx) should be treated as status `fail` and data to specify the field that should be corrected.

Comment (nickhsine) [reply controllers/donation.go]: Will change to `gin.H{status:"fail" "data": gin.H{"url": "url can not address a resource"}}`

**Comment by nickhsine** [comment]
all fixed @babygoat

**Pull request merged**